src/parsers/installrdf.js

```diff
@@ -0,0 +1,56 @@
+import { ADDON_TYPE_MAP, INSTALL_RDF } from 'const';
+import { TYPE_INVALID,
+         TYPE_MISSING } from 'messages';
+
+import log from 'logger';
+import RDFScanner from 'scanners/rdf';
+
+
+export default class InstallRdfParser {
+
+  constructor(rdfString, collector) {
+    this.rdfString = rdfString;
+    // Provides ability to directly add messages to
+    // the collector.
+    this.collector = collector;
```

Comment:
Scratch this comment. I'm looking over this again as I feel like my brain wasn't working this evening. I get what's going on now and this makes perfect sense. Sorry! :disappointed: